tests/unit/models/registration-test.ts

```diff
@@ -9,4 +9,61 @@ module('Unit | Model | registration', hooks => {
         const model = run(() => this.owner.lookup('service:store').createRecord('registration'));
         assert.ok(!!model);
     });
+
+    test('registraiton state selects public as default', function(assert) {
+        const model = run(() => this.owner.lookup('service:store').createRecord('registration'));
+
+        assert.equal(model.get('state'), 'public');
+    });
+
+    test('registration state returns pendingRegistrationApproval if field is true', function(assert) {
+        const model = run(() => this.owner.lookup('service:store').createRecord('registration'));
+        model.set('pendingRegistrationApproval', true);
+
+        assert.equal(model.get('state'), 'pendingRegistrationApproval');
+    });
+
+    test('registration state returns pendingEmbargoApproval if field is true', function(assert) {
+        const model = run(() => this.owner.lookup('service:store').createRecord('registration'));
+        model.set('pendingEmbargoApproval', true);
+
+        assert.equal(model.get('state'), 'pendingEmbargoApproval');
+    });
+
+    test('registration state return pendingEmbargoTerminationApproval if field is true', function(assert) {
```

Comment:
For this particular case, both `embargoed` and `pendingEmbargoTerminationApproval` flags are `true`, so you could also test that to make sure the return state is `pendingEmbargoTerminationApproval`.

Reply:
Wait, this would imply the order of that if function was relevant - is that true?

Reply:
If yes, are there any other double states like that?

Reply:
And just to clarify, if both embargoed and pending are true it should return pendingEmbargoTermination?

Reply:
So this actually does work, but I'm 100% sure its just sheer luck. Gonna try to get it deterministic.

Reply:
this is only true for this case. This is because, in this case, we are requesting to end an existing embargo. So yes, the order matters. I think `Object.keysOf` returns keys in same order as listed so it's deterministic so long as no one changes the order of listing. worth looking into and adding a test case

Reply:
so `embargo` related states are `pendingEmbargoApproval`, `embargoed`, `pendingEmbargoTerminationApproval`. Only the last two can be `true` at the same time as per current API deisign.

Reply:
Already added a test case, and made it deterministic. We shouldn't ever rely on the natural order objects are given us tbh this can (and most likely will) change. Maybe someone changes a name or does x, or y. I just forced the bool to be false for the Embargo when the termination approval is true.

Reply:
^ It's a recipe for a very vague and random bug to crop up

Reply:
\*sigh* I was also thinking there was only one `true` at a time. Looking more carefully, I *think* these are the only pairs that might be true at the same time:
- embargoed + pending embargo termination => pending embargo termination
- public + pending withdrawal => pending withdrawal
but it's hard to be sure because the backend logic is all disconnected.

Reply:
Its super easy to enforce this - are you 100% sure about the public + pending withdrawal? Just a one line change and a test case away from implementation.

Reply:
[Yep.](https://api.staging.osf.io/v2/registrations/m3f2r/?fields[registrations]=public,pending_withdrawal)
```
        "attributes": {
            "pending_withdrawal": true,
            "public": true
        },
```

Reply:
kewl

Reply:
Actually, this *should* work because pending_withdrawal will return if true, public is the default case.

Reply:
wrapping with tests.

Reply:
should be good to go.